### ISSUE TITLE
Detecting large object in image

### ISSUE THREAD

**Issue by jwee1369** [ultralytics/ultralytics#9521]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hi,

I have input image of 1920x1080 and the object size is 1920x200 ie. spans the width of the image with a little height.

When training with 640x360, i'm getting Box MAP 0.99 and Pose MAP 0.995

However when i train with 1920x1080, i get Box MAP 0.116 and Pose MAP 0.995

When i set reg_max=100, and train with 1920x1080, i get Box MAP 0.931, and Pose MAP 0.995 

All the above were trained for the same number of epochs.

For the Box prediction, the 640x360 (with default reg_max=20), performs the best, and the pose detection points looks better qualitatively. All this with 3M+ parameter

The 1920x1080 with reg_max=100, performs better than having reg_max=20, but the box and pose is not as good as the 640x360

I have also tried changing the box loss weight and lr0, but none had effect, is there anything else I could do to have better performance at higher resolution ?

Thanks,
Jason.

**Comment by glenn-jocher** [comment]
@jwee1369 hi Jason,

👋 It sounds like you're dealing with an interesting case of object detection using YOLOv8, focusing on large objects spanning your images. The discrepancy in performance between different resolutions and `reg_max` settings is indeed a curious scenario.

Given your observations, here are a couple of suggestions you might find useful:
- **Aspect Ratios**: Since your object spans the width with minimal height, consider adjusting the anchor aspect ratios to better match this. YOLOv8 automatically optimizes these during training, but a manual override could provide better initial bearings.
- **Feature Pyramid Networks (FPN) Layers**: Training with higher resolutions benefits from deeper FPN layers, as they can capture the fine details lost at lower resolutions. Adjusting the network depth might help capture better features for large objects.
- **Augmentation**: If not already implemented, consider leveraging extensive image augmentation techniques to ensure the model is not overfitting to specific sizes but rather learning the object features robustly.

A quick code snippet to adjust anchor aspects and augmentation, for example:

```yaml
anchors:
  - [116,90, 156,198, 373,326]  # adjusting aspect ratios to wider shapes

augmentations:
  - name: RandomResizedCrop
    p: 1.0
    scale: [0.6, 1.0]  # Ensure variety in object size
```

Increasing the dataset diversity with augmentation and fine-tuning the network to better suit large, wide objects might provide additional gains in performance at higher resolutions.

Feel free to share additional findings or raise more questions as you continue experimenting. Good luck with your training! 🚀

**Comment by Hungtran-pro** [comment]
@glenn-jocher Could you explain the meaning of number in anchor list Do they represent for width, height in pixel or anything else?

**Comment by glenn-jocher** [comment]
@Hungtran-pro the numbers in the anchor list represent the width and height of the anchor boxes in pixels, used to predict bounding boxes during object detection.
